components/HackathonModal.tsx

```diff
@@ -0,0 +1,142 @@
+import Link from 'next/link';
+import styles from '../styles/components/HackathonModal.module.scss';
+import { website_config } from '../config';
+
+const MODAL_DISPLAY_INTERVAL = 2 * 24 * 60 * 60 * 1000; // 2 days in milliseconds
+
+interface HackathonModalProps {
+  onRequestClose: () => void;
+}
+
+const HackathonModal: React.FC<HackathonModalProps> = ({ onRequestClose }) => {
+  const [isOpen, setIsOpen] = useState<boolean>(false);
+  const [assetsLoaded, setAssetsLoaded] = useState<boolean>(false);
+
+  useEffect(() => {
+    const fontFamilies = ['Orbitron', 'Poppins', 'Motorblock'];
+
+    const imageUrls = [
+      '/Hackathon/little-guy.png',
+      '/Hackathon/top-left.svg',
+      '/Hackathon/top-right.svg',
+      '/Hackathon/bottom-left.svg',
+    ];
+
+    const loadFonts = async () => {
+      try {
+        await Promise.all(
+          fontFamilies.map((family) => document.fonts.load(`1em "${family}"`))
+        );
+      } catch (error) {
+        console.error('Failed to load fonts:', error);
+      }
+    };
+
+    const loadImages = async () => {
+      try {
+        await Promise.all(
+          imageUrls.map((url) => {
+            return new Promise<void>((resolve, reject) => {
+              const img = new window.Image();
+              img.onload = () => resolve();
+              img.onerror = () =>
+                reject(new Error(`Failed to load image: ${url}`));
+              img.src = url;
+            });
+          })
+        );
+      } catch (error) {
+        console.error('Failed to load images:', error);
+      }
+    };
+
+    Promise.all([loadFonts(), loadImages()])
+      .then(() => setAssetsLoaded(true))
+      .catch((error) => console.error('Failed to load assets:', error));
+  }, []);
+
+  useEffect(() => {
+    if (assetsLoaded) {
+      const lastViewed = localStorage.getItem('lastViewed');
+      const currentTime = Date.now();
+
+      if (
+        !lastViewed ||
+        currentTime - Number(lastViewed) > MODAL_DISPLAY_INTERVAL
+      ) {
+        setIsOpen(true);
+        localStorage.setItem('lastViewed', currentTime.toString());
+      }
+    }
+  }, [assetsLoaded]);
+
+  const handleClose = () => {
+    setIsOpen(false);
+    onRequestClose();
+  };
+
+  if (!assetsLoaded) {
+    return null; // Don't render anything until assets are loaded
+  }
+
+  return (
+    <Modal
+      isOpen={isOpen}
+      onRequestClose={handleClose}
+      contentLabel='Welcome Modal'
+      className={styles.Modal}
+      overlayClassName={styles.ModalOverlay}
+    >
+      <Link href={website_config.hackathon_url}>
+        <div className={styles.modalImageContainer}>
+          <Image
+            src='/Hackathon/little-guy.png'
+            alt='Modal Background'
+            layout='fill'
+            objectFit='fit'
+          />
+        </div>
+        <div className={styles.modalContent}>
+          <div className={styles.imageContainer}>
+            <div className={styles.topLeft}>
```

Comment:
I think these should probably all be span ekements. the way I think about it is `div` indicates it has sizing and moves stuff out of the way in the box, or holds a group. Span indicates that we just need a wrapper.